### PR TITLE
Fix PUT operations: fully replace the content of a record with a new one

### PR DESCRIPTION
**Pull request by imcaizheng**
## Screenshots as Verification

1. PUT /jobs/:jobId

    updateJob

2. PUT /jobCandidates/:jobCandidateId

    updateJobCandidate

3. PUT /resourceBookings/:resourceBookingId

    updateResourceBooking

### CODE REVIEW

File: docs/swagger.yaml [hunk 1]
```diff
@@ -2003,7 +2003,6 @@ components:
       required:
         - projectId
         - userId
-        - status
```
Comment (imcaizheng) [inline docs/swagger.yaml:2006]: I believed that, to be consistent, in the `PUT` operation `status` should have a default value like in the  `POST` operation. So `status` is no longer being a required field.

Comment (maxceem) [reply docs/swagger.yaml:2006]: Good catch, agree 👍

**Comment by maxceem** [comment]
@imcaizheng PR looks good to me. Could you please, resolve the merge conflict after merging the previous PR.

**Comment by imcaizheng** [comment]
@maxceem PR updated.

**Review by maxceem — APPROVED** [review]
Thank you for so quick update @imcaizheng.
Works good as per local testing.

**Pull request merged**